website/docs/guides/middlewares/consumer-throttling-middleware.md

```diff
@@ -0,0 +1,60 @@
+---
+sidebar_position: 6
+---
+
+# Consumer Throttling
+
+Consumer Throttling is a feature in KafkaFlow that allows you to dynamically manage the rate at which your application consumes messages from Kafka topics. It provides a mechanism to slow down or pause the consumption of messages based on specified metrics. This can be particularly useful in scenarios where you want to prioritize certain types of messages over others, or when there is a need to control the rate of processing during high-load periods or when dealing with backlogs.
+
+This functionality is highly extensible, allowing for the use of custom metrics and actions by implementing the `IConsumerThrottlingMetric` and `IConsumerThrottlingAction` interfaces respectively. Although KafkaFlow comes with built-in implementations like Consumer Lag for metrics and Delay for actions, you can create your own based on your specific needs.
+
+## Use Case Example
+
+One example of how Consumer Throttling can be utilized effectively is when you want to segregate single and bulk actions into different consumers and topics. For instance, you may want to prioritize the processing of single actions and slow down the processing of bulk actions. This can be achieved by monitoring the consumer lag of the consumer responsible for single actions and applying throttling to the consumer handling the bulk actions based on this metric.
+
+## How to Configure Consumer Throttling
+
+Configuring Consumer Throttling is straightforward with the fluent interface provided by KafkaFlow. Here's a simple example:
+
+```csharp
+.AddConsumer(
+    consumer => consumer
+        .Topic("bulk-topic")
+        .WithName("bulkConsumer")
+        .AddMiddlewares(
+            middlewares => middlewares
+                .ThrottleConsumer(
```

Comment:
What about `.Throttle(` or `.Throttling(` since we are already in the context of a consumer?